spock-core/src/main/java/org/spockframework/runtime/extension/builtin/IncludeExcludeExtension.java

```diff
@@ -99,5 +104,24 @@ private boolean hasAnyBaseClass(SpecInfo spec, List<Class<?>> baseClasses) {
 
     return false;
   }
+
+  private boolean matchesAnyPattern(NodeInfo<?, ?> node, List<Pattern> patterns) {
+    if (matchesAnyPattern(node.getName(), patterns))
+      return true;
+
+    for (Annotation ann : node.getAnnotations())
+      if (matchesAnyPattern(ann.annotationType().getSimpleName(), patterns))
```

Comment:
Please use the fully qualified class name instead to avoid collisions with test names.